### ISSUE TITLE
critical: wrong sequences shown on litter page

### ISSUE THREAD

**Issue by nsteinme** [cortex-lab/alyx#475]
Hi, just discussed a critical issue with @charureddy : 

Genotype test results are entered from the litter page, where it shows up to three sequences and lets you quickly put in the test results. However, the wrong sequences are being shown. E.g. see TetO.CtTa_L_081, which shows sequences tTA (correct), VSFP (wrong), and eGFP (wrong). It is supposed to also show sequence TRE but incorrectly omits it. This is a critical issue because it is impractical to enter all the genotype test information for sequence TRE subject-by-subject from the subject pages (where you can pick any sequence for any subject, so the info can be entered in principle there). Because it is impractical, recent information has not been entered. Note that we have confirmed the test results are not lost or missing - older mice do have test results for those sequences. But they don't show up on litter pages. 

I don't actually remember where/how these "line-sequence relationships" are determined or specified - I don't think there is a table for it like there is for line-allele relationships. It could get it from the zygosity rules table, but it seems that it doesn't, or else there is a bug with that. Please let me know if you need clarification! Would appreciate resolution asap. 

https://alyx.cortexlab.net/admin/subjects/litter/cdee2adf-034b-44bc-a15b-aa79bbaaa0fe/change/?_changelist_filters=line__id__exact%3D6a177a19-4787-461f-932b-a93cffe55f7c%26p%3D1

**Comment by nsteinme** [comment]
Great, thanks, we had missed the allele-sequence relationships. Some of these were set incorrectly (e.g. VSFP was for some reason associated with the tetO-g6s allele) which was causing the problem. However I think at least in principle we probably do need a line-sequence relationship. Here's why. 

In the case of a single transgenic line, you frequently want a genotype test on the transgene and on the wild-type sequence that should be found when the transgene is not present at all - this allows you to tell whether it is homo/het/absent rather than just present/absent. But when you cross that single transgenic line with another single transgenic line, for these offspring you just care about present/absent so you won't do the wild-type sequence tests ( anyway you know WT will always be positive since the subjects can't be homozygous). The problem arises because you now have four sequences legitimately associated with those two alleles - so there will be no way to pick which three to show on the litter page. 

So the upshot is we probably do need some way to specify directly which one shows up on the litter page. That's *in principle* - I'm not sure whether there are actually any situations today where that's a problem, once we fix the allele-sequence relationships. It may be that in practice, all cases come up with just three sequences. We'll have to check.

**Comment by rossant** [comment]
An alternative is to show more than 3 sequences, or to find a more sensible way of picking the order these sequences are displayed on the page.

**Comment by nsteinme** [comment]
yes, I think either of those would work but picking the order is more future-proof probably.
